core/audits/redirects.js

```diff
@@ -63,18 +64,19 @@ class Redirects extends Audit {
       const data = event.args.data || {};
       if (!data.documentLoaderURL || !data.isLoadingMainFrame) continue;
 
-      let networkRecord = networkRecords.find(record => record.url === data.documentLoaderURL);
+      let networkRecord = networkRecords.find(record => record.requestId === data.navigationId);
       while (networkRecord) {
         documentRequests.push(networkRecord);
+        // HTTP redirects won't have separate navStarts, so find through the redirect chain.
         networkRecord = networkRecord.redirectDestination;
       }
     }
 
-    // If we found documents in the trace, just use this directly.
-    if (documentRequests.length) return documentRequests;
+    if (!documentRequests.length) {
+      throw new Error('No navigation requests found');
+    }
 
-    // Use the main resource as a backup if we didn't find any modern navigationStart events
-    return (mainResource.redirects || []).concat(mainResource);
```

Comment:
this is for super ancient traces (like m60, maybe), and there are no tests hitting this, so no need to keep it around.